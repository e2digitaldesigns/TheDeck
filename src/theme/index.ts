export * from "./styles/forms.styles";
export * from "./styles/scrollBars.styles";
