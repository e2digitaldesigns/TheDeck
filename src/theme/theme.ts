export const theme = {
  sizes: {
    application: {
      height: "785px",
      width: "1280px"
    },
    header: {
      height: "2rem",
      width: "100vw"
    },
    macroDeck: {
      sideBar: { width: "240px" },
      mainContent: { width: "1016px" }
    }
  },
  colors: {
    body: "black",
    border: "#111",
    accent1: "#f542e9",
    accent2: "#291697",
    accent3: "#404040",
    accent4: "#2b2b2b",
    accent5: "#1a1a1a",
    gridBg: "43, 45, 48,",
    font: "white",
    main: "#100727",
    mainBackground: "black",
    scrollBackground: "#100727",
    scrollFont: "#dedede",
    scrollHeader: "black",
    scrollHeaderFont: "white"
  }
};
