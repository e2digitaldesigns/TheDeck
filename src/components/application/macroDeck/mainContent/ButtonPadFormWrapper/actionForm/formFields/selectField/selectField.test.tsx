import React from "react";
import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";
import FormFieldSelect, { IntFormFieldSelectProps } from "./selectField";

import { ThemeProvider } from "styled-components";
import { theme } from "../../../../../../../../theme/theme";

const mocked = {
  name: "md",
  onChange: jest.fn(),
  state: {
    profileId: "profileId",
    pageId: "pageId",
    buttonPadId: "buttonPadId",
    _id: "_id",
    order: 2,
    action: "",
    subAction: "mdReset",
    seconds: 0,
    url: "",
    text: "b",
    scene: "",
    layer: "",
    page: "",
    path: "",
    profile: ""
  }
};

const defaultProps = {
  name: mocked.name,
  onChange: mocked.onChange,
  subAction: mocked.state.subAction
};

const testSetup = (props: IntFormFieldSelectProps) => {
  return render(
    <ThemeProvider theme={theme}>
      <FormFieldSelect
        name={props.name}
        onChange={props.onChange}
        subAction={props.subAction}
      />
    </ThemeProvider>
  );
};

let wrapper: any = null;
beforeEach(() => {
  wrapper = testSetup(defaultProps);
});

afterEach(() => {
  wrapper = null;
  jest.clearAllMocks();
});

describe("<FormFieldSelect/>", () => {
  it("Should render without errors", () => {
    const component = testSetup(defaultProps);
    expect(component).toMatchSnapshot();
  });

  it("Should call onChange", () => {
    const input = wrapper.getByTestId("form_field_selectField__input");
    expect(input).toBeTruthy();

    const value = "mdSettings";
    fireEvent.change(input, { target: { value } });
    expect(mocked.onChange).toHaveBeenCalled();
  });
});
