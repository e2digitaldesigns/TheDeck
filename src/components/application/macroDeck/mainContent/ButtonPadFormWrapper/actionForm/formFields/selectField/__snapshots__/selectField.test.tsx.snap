// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormFieldSelect/> Should render without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <label
          data-testid="form_field_selectField__label"
          for="md"
        >
          Md
          :
        </label>
        <select
          class="sc-dkPtRN iRRDCl"
          data-testid="form_field_selectField__input"
          name="subAction"
        >
          <option
            data-testid="form_field_selectField__option-choose"
            value=""
          >
            Choose
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdHome"
          >
            mdHome
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdPage"
          >
            mdPage
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdProfile"
          >
            mdProfile
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdProfileSelector"
          >
            mdProfileSelector
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdReset"
          >
            mdReset
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdSettings"
          >
            mdSettings
          </option>
        </select>
      </div>
    </div>
    <div>
      <div>
        <label
          data-testid="form_field_selectField__label"
          for="md"
        >
          Md
          :
        </label>
        <select
          class="sc-dkPtRN iRRDCl"
          data-testid="form_field_selectField__input"
          name="subAction"
        >
          <option
            data-testid="form_field_selectField__option-choose"
            value=""
          >
            Choose
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdHome"
          >
            mdHome
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdPage"
          >
            mdPage
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdProfile"
          >
            mdProfile
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdProfileSelector"
          >
            mdProfileSelector
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdReset"
          >
            mdReset
          </option>
          <option
            data-testid="form_field_selectField__option"
            value="mdSettings"
          >
            mdSettings
          </option>
        </select>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <label
        data-testid="form_field_selectField__label"
        for="md"
      >
        Md
        :
      </label>
      <select
        class="sc-dkPtRN iRRDCl"
        data-testid="form_field_selectField__input"
        name="subAction"
      >
        <option
          data-testid="form_field_selectField__option-choose"
          value=""
        >
          Choose
        </option>
        <option
          data-testid="form_field_selectField__option"
          value="mdHome"
        >
          mdHome
        </option>
        <option
          data-testid="form_field_selectField__option"
          value="mdPage"
        >
          mdPage
        </option>
        <option
          data-testid="form_field_selectField__option"
          value="mdProfile"
        >
          mdProfile
        </option>
        <option
          data-testid="form_field_selectField__option"
          value="mdProfileSelector"
        >
          mdProfileSelector
        </option>
        <option
          data-testid="form_field_selectField__option"
          value="mdReset"
        >
          mdReset
        </option>
        <option
          data-testid="form_field_selectField__option"
          value="mdSettings"
        >
          mdSettings
        </option>
      </select>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
